core/src/macosX64Main/kotlin/Peripheral.kt

```diff
@@ -130,9 +131,12 @@ public class ApplePeripheral internal constructor(
                 _connection.value = it
             }
 
-            connection.characteristicChanges
+            connection
+                .delegate
+                .characteristicChanges
+                .takeWhile { it !== Closed }
```

Comment:
I don't know what type Closed is, but did you intentionally use the referential operator? (!==)

Reply:
`Closed` is a "singleton" `object`, so the `!==` is a small optimization for that comparison. `!=` would've worked (slightly less efficient though).

https://github.com/JuulLabs/kable/blob/09619e55d9cdd8e62edad873c1d916bdd5aa1fab/core/src/macosX64Main/kotlin/PeripheralDelegate.kt#L79-L93

Reply:
Got it. Thanks.